### ISSUE TITLE
Getting errors on XHTML declaration

### ISSUE THREAD

**Issue by GeorgeKerscher** [htmlhint/vscode-htmlhint#203]
I have valid XHTML files and I get three warnings in the first two lines. They are:
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" lang="en">

It seems that the rest of the wornings work the same if I am in XHTML or HTML. It works well.

It would be good if I could not get these warnings. Right now, I just ignore the warnings.

Is there something in configuration that could resolve this issue? I have looked through the seetings, but do not find anything that applies.

FYI I am blind and using a screen reader, so I may be missing something. Also fairly new to VS Code and I am loving it.

**Comment by coliff** [comment]
Hi George! Thanks for reporting this! I think you had intended to include a code snippet or screenshot but there isn't one. What are the three warnings in the first two lines you mentioned? Once you let me know about that I'll be able to advise. Look forward to hearing back from you.

**Comment by GeorgeKerscher** [comment]
Here are the three errors followed by the first three lines of the XHTML file.
level 1  3 problems in file introduction.xhtml of folder C:\Users\kersc\OneDrive - DAISY Consortium\work\EPUB always up-to-date\Fundamental-Accessibility-Tests-Non-Visual-Reading-v2.0.0-unpacked\EPUB\xhtml  not selected  expanded  1 of 1
level 2  Warning: Special characters must be escaped : [ < ]. at line 1 and character 1.  1 of 3
Warning: Doctype must be declared first. at line 1 and character 1.  2 of 3  level 2
Warning: Special characters must be escaped : [ > ]. at line 1 and character 38.  3 of 3  level 2

Here are the first few lines of the file:
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">

<head>

**Comment by GeorgeKerscher** [comment]
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">

Avove are the first three lines of the file.

I also edit HTML and would not want to impact that functionality. Would:
In any case, if you are using XHTML then you will likely need the following in your .htmlhintrc config.
  "doctype-first": false,
  "doctype-html5": false,

Sould that impact my HTML editing?

**Comment by coliff** [comment]
That will not impact your HTML editing at all.

I use HTMLHint in many different kinds of projects and always have a custom `.htmlhintrc` config in the root of my project with rules for the type of project. It's best to disabled the rules which don't apply to you so they don't get in the way when editing in VS Code.

**Comment by GeorgeKerscher** [comment]
First, thank you for the help.

The doctype set to false removed one error. I still have two in the first line which is:
  <?xml version="1.0" encoding="UTF-8"?>

The warnings
Warning: Special characters must be escaped : [ < ]. at line 1 and character 1.  
Warning: Special characters must be escaped : [ > ]. at line 1 and character 38

I have spent a lot of time going through documentation and cannot figure out what to do to make this warning go away.

I would think that this declaration would be a fairly common request.

Any suggestions welcome.

**Comment by coliff** [comment]
For that error you can use this in the HTMLHint config:

"spec-char-escape": false,

**Comment by GeorgeKerscher** [comment]
It seems this would allow < dangling anywhere. I don't think we want that. I am just trying to get the first line of an XHTML file to be without errors.

XML is the file format used in EPUB, which is being used by publishers all over the world. Publishers freek out when they see warnings in documents they are working on. I would love to see an elegant solution.

**Comment by coliff** [comment]
You haven't shared the code that was problematic with me yet. If you can do that then that would help me to suggest a solution.

**Comment by GeorgeKerscher** [comment]
Hi,

I am just looking for a rule or two that will allow the first line of an XHTML file not to throw errors. That is:
<?xml version="1.0" encoding="utf-8"?>

The rest of HTMLHINT rules work fine.

**Comment by coliff** [comment]
Ah, I realized what was happening.. the code you had pasted in your reply was hidden (invisible) in the GitHub comments UI so I didn't see the `<?xml version="1.0" encoding="utf-8"?>` you had previously pasted. Sorry about that - I just discovered it by editing your post to see the raw markdown code!

The code you provided in your first post was:

```html
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" lang="en">
```
However, that doesn't look right for XHTML. I think it should be:

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
"http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
```

The `<!DOCTYPE html>` is for HTML5, not XHTML.
I don't think the XML declaration (`<?xml version="1.0" encoding="UTF-8"?>`) is ever needed for XHTML... I suggest removing that.
